grails-app/views/purchaseOrderWorkflow/purchaseOrder/showOrderItems.gsp

```diff
@@ -337,6 +337,51 @@
           return false
         }
 
+        function cancelOrderItem(id) {
```

Comment:
This and restoreOrderItem could be one function, that gets params with id, action, and error message. Other than that lgtm